src/Symfony/Component/Security/Core/Authorization/Voter/AbstractVoter.php

```diff
@@ -146,6 +139,6 @@ protected function isGranted($attribute, $object, $user = null)
      */
     protected function voteOnAttribute($attribute, $object, TokenInterface $token)
     {
-        return false;
+        return $this->isGranted($attribute, $object, $token->getUser());
```

Comment:
Can we not move the `trigger_error` here as well? Then we don't need reflection at all.